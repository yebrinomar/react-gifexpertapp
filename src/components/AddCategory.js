import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e)=>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        console.log('Submit hecho');
        
        if(inputValue.trim().length > 2){
            setCategories(cats => [ inputValue, ...cats ]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}             
            />
        </form>       
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

