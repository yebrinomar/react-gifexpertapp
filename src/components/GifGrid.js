//import React, { useState, useEffect } from 'react'
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    //const [images, setImages] = useState([]);
    const {data:images, loading} = useFetchGifs(category);

    // useEffect( ()=> {
    //     getGifs(category)
    //     .then(setImages);
    // }, [category])

    return ( 
        <>
            <h3 className="card animate__animated animate__fadeIn"> {category} </h3>  
            {loading && <p className="card animate__animated animate__flash">Loading</p>}
            <div className="card-grid">

                {
                    images.map( img => {
                        return <GifGridItem  
                            key={img.id}
                            {...img}
                            //img = {img} 
                        />
                    })
                }
            </div>
        </>
       
    )
}
