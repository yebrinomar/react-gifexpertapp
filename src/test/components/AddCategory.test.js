import React from 'react';
import {shallow} from 'enzyme';
import {AddCategory} from '../../components/AddCategory'

describe('Prueba en el <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = {setCategories}/>);;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = {setCategories}/>);
    });

    test('debe de mostrar <AddCategory /> correctamente', () => {
 
        expect(wrapper).toMatchSnapshot();
    
    });

    
    test('debe de cambiar la caja de texto', () => {
 
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: {  value } });
        expect(wrapper.find('p').text().trim()).toBe(value);
    
    });

    test('NO debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){ } });     
        expect( setCategories ).not.toHaveBeenCalled();
        
    }); 

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        // 1. Simular el inputChange
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: {  value } });

        // 2. simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // 3. setCategories se debe haber llamado
        expect( setCategories ).toHaveBeenCalled();  
        expect( setCategories ).toHaveBeenCalledTimes(1);  
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function)); 

        // 4. el valor del input debe estar en ''
        expect(wrapper.find('input').prop('value')).toBe('');
        
    }); 
})
