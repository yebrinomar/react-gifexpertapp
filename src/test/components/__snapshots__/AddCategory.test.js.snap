// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el <AddCategory /> debe de mostrar <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
